docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp.rst

```diff
@@ -1,14 +1,19 @@
 .. _analyzing-data-flow-in-cpp:
 
-.. pull-quote:: Note
-
-   The data flow library used in this article has been replaced with an improved library which is available from CodeQL 2.12.5 onwards, see :ref:`Analyzing data flow in C and C++ (new) <analyzing-data-flow-in-cpp-new>`. The old library has been deprecated in CodeQL 2.14.1 and will be removed in a later release. With the release of CodeQL 2.13.0 both libraries use the new modular API for data flow.
-
 Analyzing data flow in C and C++
 ================================
 
 You can use data flow analysis to track the flow of potentially malicious or insecure data that can cause vulnerabilities in your codebase.
 
+About this article
+------------------
+
+This article describes how data flow analysis is implemented in the CodeQL libraries for C/C++ and includes examples to help you write your own data flow queries.
+The following sections describe how to use the libraries for local data flow, global data flow, and taint tracking.
+For a more general introduction to modeling data flow, see ":ref:`About data flow analysis <about-data-flow-analysis>`."
```

Comment:
This link is now duplicated in the "About data flow" section a few lines below.  I think we only need one or the other.

Reply:
```suggestion
```